workflow/workflow.go

```diff
@@ -33,6 +30,24 @@ func (w *Workflow) Validate() error {
 	return nil
 }
 
+func (e *Edge) validate(w *Workflow) error {
```

Comment:
that's a weird function. An edge that accept the workflow? why moving it to a new function anyway?

Reply:
just wanted to move it because this validation will be more and more complex with validation of the references, inputs/outputs etc... but I agree maybe `func (w Workflow) validateEdge(e Edge)` would be better